tests/dispatch/test_http2.py

```diff
@@ -133,16 +201,27 @@ def test_http2_multiple_requests():
     assert json.loads(response_3.content) == {"method": "GET", "path": "/3", "body": ""}
 
 
+def mock_app(request):
+    content = json.dumps(
+        {
+            "method": request.method,
+            "path": request.url.path,
+            "body": request.content.decode(),
+        }
+    ).encode()
+    headers = {"Content-Length": str(len(content))}
+    return Response(200, headers=headers, content=content)
+
+
 def test_http2_reconnect():
     """
     If a connection has been dropped between requests, then we should
     be seemlessly reconnected.
     """
-    backend = MockHTTP2Backend(app=app)
+    backend = MockHTTP2Backend(app=mock_app)
 
     with Client(backend=backend) as client:
         response_1 = client.get("http://example.org/1")
-        backend.server.close_connection = True
```

Comment:
Woops, will restore.

Reply:
I think using the mock backend here is probably better than Hypercorn as it is hard to make the latter drop a connection from the outside.